lib/actions/user.js

```diff
@@ -181,3 +188,91 @@ export function deleteUserMonitoredTrip (id) {
     }
   }
 }
+
+/**
+ * Requests a verification code via SMS for the logged-in user.
+ */
+export function requestPhoneVerificationSms (originalUserData, newPhoneNumber) {
+  return async function (dispatch, getState) {
+    const { otp, user } = getState()
+    const { otp_middleware: otpMiddleware = null } = otp.config.persistence
+
+    if (otpMiddleware) {
+      const { accessToken } = user
+
+      // FIXME: (required by middleware) Only temporarily update the user's phone number
+      //   in the database with the pending one,
+      //   and cache the previous value and verification state so they can be reverted.
+      //   The full user record will be updated upon the user clicking Finish/Save preferences.
+      // TODO: Figure out what should happen if the user refreshes the browser at this stage.
+
+      // Make a clone of the original userData object.
+      const newUserData = clone(originalUserData)
+      const previousPhoneNumber = newUserData.phoneNumber
+      const previousIsPhoneNumberVerified = newUserData.isPhoneNumberVerified
+
+      newUserData.phoneNumber = newPhoneNumber
+      newUserData.isPhoneNumberVerified = false
+
+      const userUpdateResult = await updateUser(otpMiddleware, accessToken, newUserData)
+
+      if (userUpdateResult.status === 'success' && userUpdateResult.data) {
```

Comment:
This method body contains too much logic to not have any comments describing what's going on here.